Test failures with Python 3.11

In [Nixpkgs](https://github.com/NixOS/nixpkgs) we're starting to build packages for Python 3.11, but our validobj package fails its tests.  Here are the failures I found testing in the python:3.11 Docker image (to make sure it's not a problem with our setup):

```
============================= test session starts ==============================
platform linux -- Python 3.11.2, pytest-7.2.1, pluggy-1.0.0
rootdir: /validobj
plugins: hypothesis-6.68.1, cov-4.0.0
collected 19 items

validobj/tests/test_custom.py ...                                        [ 15%]
validobj/tests/test_errors.py ......                                     [ 47%]
validobj/tests/test_validation.py FF..F.....                             [100%]

=================================== FAILURES ===================================
________________________________ test_good_inp _________________________________

    def test_good_inp():
        expected_res = Db(
            metadata=good_inp['metadata'],
            flags={Flag('f1'), Flag('f2')},
            rows=[Row('Hello', (1, 2, 3)), Row('World', (4, 5, 6))],
            size=MemOptions.MEDIUM,
            attributes=Attributes.READ | Attributes.WRITE | Attributes.EXECUTE,
        )
>       assert parse_input(good_inp, Db) == expected_res

validobj/tests/test_validation.py:110:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/usr/local/lib/python3.11/site-packages/validobj/validation.py:432: in parse_input
    return _parse_dataclass(value, spec)
/usr/local/lib/python3.11/site-packages/validobj/validation.py:212: in _parse_dataclass
    res[k] = parse_input(v, field_dict[k])
/usr/local/lib/python3.11/site-packages/validobj/validation.py:430: in parse_input
    return _parse_enum(value, spec)
/usr/local/lib/python3.11/site-packages/validobj/validation.py:272: in _parse_enum
    return _parse_list_enum(value, spec)
/usr/local/lib/python3.11/site-packages/validobj/validation.py:259: in _parse_list_enum
    res = spec._create_pseudo_member_(0)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

cls = <flag 'Attributes'>, name = '_create_pseudo_member_'

    def __getattr__(cls, name):
        """
        Return the enum member matching `name`

        We use __getattr__ instead of descriptors or inserting into the enum
        class' __dict__ in order to support `name` and `value` being both
        properties for enum members (which live in the class' __dict__) and
        enum members themselves.
        """
        if _is_dunder(name):
            raise AttributeError(name)
        try:
            return cls._member_map_[name]
        except KeyError:
>           raise AttributeError(name) from None
E           AttributeError: _create_pseudo_member_

/usr/local/lib/python3.11/enum.py:789: AttributeError
_________________________________ test_bad_inp _________________________________

    def test_bad_inp():
        for k in bad_inp:
            with pytest.raises(ValidationError):
>               parse_input(*k)

validobj/tests/test_validation.py:116:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/usr/local/lib/python3.11/site-packages/validobj/validation.py:430: in parse_input
    return _parse_enum(value, spec)
/usr/local/lib/python3.11/site-packages/validobj/validation.py:272: in _parse_enum
    return _parse_list_enum(value, spec)
/usr/local/lib/python3.11/site-packages/validobj/validation.py:259: in _parse_list_enum
    res = spec._create_pseudo_member_(0)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

cls = <flag 'Attributes'>, name = '_create_pseudo_member_'

    def __getattr__(cls, name):
        """
        Return the enum member matching `name`

        We use __getattr__ instead of descriptors or inserting into the enum
        class' __dict__ in order to support `name` and `value` being both
        properties for enum members (which live in the class' __dict__) and
        enum members themselves.
        """
        if _is_dunder(name):
            raise AttributeError(name)
        try:
            return cls._member_map_[name]
        except KeyError:
>           raise AttributeError(name) from None
E           AttributeError: _create_pseudo_member_

/usr/local/lib/python3.11/enum.py:789: AttributeError
_____________________________ test_arbitrary_valid _____________________________

    @given(dbstrat)
>   def test_arbitrary_valid(db):

validobj/tests/test_validation.py:159:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
validobj/tests/test_validation.py:160: in test_arbitrary_valid
    parse_input(db, Db)
/usr/local/lib/python3.11/site-packages/validobj/validation.py:432: in parse_input
    return _parse_dataclass(value, spec)
/usr/local/lib/python3.11/site-packages/validobj/validation.py:212: in _parse_dataclass
    res[k] = parse_input(v, field_dict[k])
/usr/local/lib/python3.11/site-packages/validobj/validation.py:430: in parse_input
    return _parse_enum(value, spec)
/usr/local/lib/python3.11/site-packages/validobj/validation.py:272: in _parse_enum
    return _parse_list_enum(value, spec)
/usr/local/lib/python3.11/site-packages/validobj/validation.py:259: in _parse_list_enum
    res = spec._create_pseudo_member_(0)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

cls = <flag 'Attributes'>, name = '_create_pseudo_member_'

    def __getattr__(cls, name):
        """
        Return the enum member matching `name`

        We use __getattr__ instead of descriptors or inserting into the enum
        class' __dict__ in order to support `name` and `value` being both
        properties for enum members (which live in the class' __dict__) and
        enum members themselves.
        """
        if _is_dunder(name):
            raise AttributeError(name)
        try:
            return cls._member_map_[name]
        except KeyError:
>           raise AttributeError(name) from None
E           AttributeError: _create_pseudo_member_
E           Falsifying example: test_arbitrary_valid(
E               db={'metadata': {},
E                'flags': [],
E                'rows': [],
E                'size': 'SMALL',
E                'attributes': ['READ', 'WRITE'],
E                'connection': 'connection'},
E           )

/usr/local/lib/python3.11/enum.py:789: AttributeError

---------- coverage: platform linux, python 3.11.2-final-0 -----------
Coverage XML written to file coverage.xml

=========================== short test summary info ============================
FAILED validobj/tests/test_validation.py::test_good_inp - AttributeError: _cr...
FAILED validobj/tests/test_validation.py::test_bad_inp - AttributeError: _cre...
FAILED validobj/tests/test_validation.py::test_arbitrary_valid - AttributeErr...
========================= 3 failed, 16 passed in 1.30s =========================
```

--- a/validobj/validation.py
+++ b/validobj/validation.py
@@ -255,8 +255,7 @@ def _parse_single_enum(value, spec):
 
 
 def _parse_list_enum(value, spec):
-    # This is a hidden function to create an enum composition with no members
-    res = spec._create_pseudo_member_(0)
+    res = spec(0)
     for i, item in enumerate(value):
         try:
             res |= _parse_single_enum(item, spec)

